src/interlocked_hl.c

```diff
@@ -151,10 +165,20 @@ IMPLEMENT_MOCKABLE_FUNCTION(, INTERLOCKED_HL_RESULT, InterlockedHL_WaitForNotVal
             /* Codes_SRS_INTERLOCKED_HL_42_004: [ When waiting for the value at address to change, the milliseconds argument value shall be used as timeout. ]*/
             if (!wait_on_address(address, current_value, milliseconds))
```

Comment:
>milliseconds [](http://example.com/codeflow?start=57&length=12)

There is a general comment here for the old code that this  timeout is not properly respected.

Example:
- Let's say we called wait_on_address with 2000 ms as timeout
- wait_on_address returns as a spurious wake after 1000 ms
- we have 1000 left of the timeout, but we're going to call again wait_on_address with 2000.
- and then wait_on_address times out after these 2000ms

So in total we waited 3000 ms, not the 2000 we wanted.

Probably a separate task also.

Reply:
@dcristoloveanu  This is #7120375 which I mentioned originally for the read-and-wait design. I assigned it to Riddhi and put it in this sprint so we don't forget to also take care of that.